### PR TITLE
Adds communication subsystem

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
add: Communication consoles now share cooldowns.
/:cl:

Sharing cooldowns is good, having a central place to make announcements
is good. This moves some of the machinery for announcements out of the
consoles, which coincides with some plans I have for the communications
machinery of the station anyway.

Also, AIs make vox announcements. Need to clarify that.

### CODE REVIEW

**Comment by ExcessiveUseOfCobblestone** [comment]
I think you may be better off defining and undefining in the subsystem since that's the only place where it's called.

**Comment by optimumtact** [comment]
Isn't this a feature

File: code/__DEFINES/misc.dm [hunk 1]
```diff
@@ -623,3 +623,6 @@ var/global/list/ghost_others_options = list(GHOST_OTHERS_SIMPLE, GHOST_OTHERS_DE
 #define INTENT_HOTKEY_RIGHT "right"
 
 #define DEFAULT_AI_LAWID "default"
+
+#define COMMUNICATION_COOLDOWN 600
+#define COMMUNICATION_COOLDOWN_AI 600
```
Comment (ChangelingRain) [inline code/__DEFINES/misc.dm]: Defines are good but by god find somewhere other than misc

**Comment by coiax** [comment]
@optimumtact it's minor refactoring and fixing an issue where they have separate cooldowns, which is counterintuitive. If you're uncomfortable, then freeze tag it. I don't mind either way.

**Comment by YakumoChen** [comment]
Dislike this, it's a feature to allow AIs to announce stuff with multiple consoles.

File: code/controllers/subsystem/communications.dm [hunk 2]
```diff
@@ -0,0 +1,32 @@
+var/datum/subsystem/communications/SScommunications
+
+/datum/subsystem/communications
+	name = "Communications"
+	priority = 50
```
Comment (MrStonedOne) [inline code/controllers/subsystem/communications.dm]: Priority doesn't matter if it doesn't fire. delet dis

File: code/__DEFINES/misc.dm [hunk 3]
```diff
@@ -399,3 +399,25 @@ var/global/list/ghost_others_options = list(GHOST_OTHERS_SIMPLE, GHOST_OTHERS_DE
 #define CLOCK_SILICONS 22
 #define CLOCK_PROSELYTIZATION 23
 #define SHUTTLE_HIJACK 24
+
+//For SSTimer
+
+#define TIMER_NORMAL "normal"
+#define TIMER_UNIQUE "unique"
+
+// Ventcrawling
```
Comment (optimumtact) [inline code/__DEFINES/misc.dm]: @coiax These defines were moved, why have you added them back?

Comment (coiax) [inline code/__DEFINES/misc.dm]: Ah, must have readded them when I was solving map conflicts. I will remove them.

**Comment by MrStonedOne** [comment]
Feature.

**Comment by coiax** [comment]
Refactor.

**Comment by MrStonedOne** [comment]
Clearly says in op that it adds something.

**Comment by optimumtact** [comment]
This doesn't change any existing functionality or add new functionality MrStonedOne, so while it does refactor the comms code into a subsystem it doesn't really meet the smell test for an actual feature

**Comment by optimumtact** [comment]
stale, update the defines and ask for a reopen

**Comment by coiax** [comment]
@ChangelingRain @optimumtact reopen plz

**Comment by optimumtact** [comment]
I cant

**Comment by optimumtact** [comment]
You force pushed or recreatd teh branch, you'll have to make a new pr

**Pull request closed without merging**